complete restructuring of the project

P6rguVyrst opened the pull request:
I'm not completely happy with the result, but I think it feels a little better to read than before. 
 
* Added few basic tests to be test original functionality and imports.
* Structured the project to make it more easily comprehensible. 
* simplified "play", "synth" and "sample" functions to use list comprehensions and locals()
* Improved .gitignore
* Fixed file permissions from 755 to 644
* Removed license information from main file - now just in LICENSE file.

I'm feeling confident I left existing functionality intact, although it might be good if someone else tested it as well. If my commit breaks anything I'm willing to fix it ASAP. It feels wrong to use import *, but I didn't want to break anything.

In another branch preserve-history tried to make changes to psonic.py file more visible, but unfortunately it wasn't any better.

P6rguVyrst committed:
Fixed file permissions and added some structure. No implementation changes.

P6rguVyrst committed:
Aded test for original functionality

P6rguVyrst committed:
Separated samples and loops from psonic.py

P6rguVyrst committed:
Since changed original files, increased version number.

P6rguVyrst committed:
Bit by bit, trying to make the codebase comprehensible

P6rguVyrst committed:
simplified "play" function.

P6rguVyrst committed:
simplified "synth" function.

P6rguVyrst committed:
separated responsibilities, simplified a little, added tests for imports.

P6rguVyrst committed:
Placeholder for synth_server

P6rguVyrst committed:
Moved synth_server to separate file.

P6rguVyrst committed:
Fixed a bug that prevented playing samples

P6rguVyrst committed:
Increased version nr.

P6rguVyrst committed:
improved gitignore

P6rguVyrst committed:
removed reclessly added functionality, tried to clean up a little.

P6rguVyrst committed:
Tried to clean up a little.

gkvoelkl commented:
Hello Toomas,

I did some tests and it works fine.

Only the run function is not available any longer.
It is used in the OSC Communication example.

run(“some ruby”)
send_message

Another point:
As far as I know should be the license in every source file. What do you mean?

Regards

Gerhard

Von meinem iPad gesendet

> Am 26.02.2018 um 20:23 schrieb Toomas Ormisson <notifications@github.com>:
> 
> I'm not completely happy with the result, but I think it feels a little better to read than before.
> 
> Added few basic tests to be test original functionality and imports.
> Structured the project to make it more easily comprehensible.
> simplified "play", "synth" and "sample" functions to use list comprehensions and locals()
> Improved .gitignore
> Fixed file permissions from 755 to 644
> Removed license information from main file - now just in LICENSE file.
> I'm feeling confident I left existing functionality intact, although it might be good if someone else tested it as well. If my commit breaks anything I'm willing to fix it ASAP. It feels wrong to use import *, but I didn't want to break anything.
> 
> In another branch preserve-history tried to make changes to psonic.py file more visible, but unfortunately it wasn't any better.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/gkvoelkl/python-sonic/pull/17
> 
> Commit Summary
> 
> Fixed file permissions and added some structure. No implementation changes.
> Aded test for original functionality
> Separated samples and loops from psonic.py
> Since changed original files, increased version number.
> Bit by bit, trying to make the codebase comprehensible
> simplified "play" function.
> simplified "synth" function.
> separated responsibilities, simplified a little, added tests for imports.
> Placeholder for synth_server
> Moved synth_server to separate file.
> Fixed a bug that prevented playing samples
> Increased version nr.
> improved gitignore
> removed reclessly added functionality, tried to clean up a little.
> Tried to clean up a little.
> File Changes
> 
> A .gitignore (64)
> M LICENSE (0)
> M README.rst (0)
> D psonic.py (950)
> A psonic/__init__.py (1)
> A psonic/api.py (84)
> A psonic/effects.py (20)
> A psonic/internals/__init__.py (0)
> A psonic/internals/chords.py (74)
> A psonic/internals/scales.py (87)
> A psonic/notes.py (73)
> A psonic/psonic.py (160)
> A psonic/samples/__init__.py (6)
> A psonic/samples/ambient.py (15)
> A psonic/samples/bass.py (13)
> A psonic/samples/drums.py (45)
> A psonic/samples/electric.py (30)
> A psonic/samples/guitars.py (9)
> A psonic/samples/loops/__init__.py (14)
> A psonic/samples/misc.py (8)
> A psonic/samples/percussions.py (10)
> A psonic/samples/tabla.py (31)
> A psonic/samples/vinyl.py (8)
> A psonic/scales.py (81)
> A psonic/synth_server.py (111)
> A psonic/synthesizers.py (51)
> M python-sonic.ipynb (0)
> M setup.py (46)
> A tests/__init__.py (0)
> A tests/conftest.py (0)
> A tests/test_psonic.py (52)
> Patch Links:
> 
> https://github.com/gkvoelkl/python-sonic/pull/17.patch
> https://github.com/gkvoelkl/python-sonic/pull/17.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

P6rguVyrst committed:
Fixed a bug stupidly introduced by refactoring, added test for it.

P6rguVyrst committed:
minor fixes

P6rguVyrst committed:
Added missing osc communication tests.

P6rguVyrst commented:
Hey!

>Only the run function is not available any longer.
>It is used in the OSC Communication example.

Sorry about that, introduced that bug by renaming something I shouldn't
have. It's fixed now and visible in PR.  While fixing, noticed few style
errors I had made and fixed those as well.


>As far as I know should be the license in every source file. What do you
mean?

It's a judgement call:
https://stackoverflow.com/questions/845895/putting-license-in-each-code-file
urllib3, for example, uses MIT license: https://github.com/shazow/urllib3




On Sun, Mar 4, 2018 at 6:06 PM, Gerhard Völkl <notifications@github.com>
wrote:

> Hello Toomas,
>
> I did some tests and it works fine.
>
> Only the run function is not available any longer.
> It is used in the OSC Communication example.
>
> run(“some ruby”)
> send_message
>
> Another point:
> As far as I know should be the license in every source file. What do you
> mean?
>
> Regards
>
> Gerhard
>
> Von meinem iPad gesendet
>
> > Am 26.02.2018 um 20:23 schrieb Toomas Ormisson <notifications@github.com
> >:
> >
> > I'm not completely happy with the result, but I think it feels a little
> better to read than before.
> >
> > Added few basic tests to be test original functionality and imports.
> > Structured the project to make it more easily comprehensible.
> > simplified "play", "synth" and "sample" functions to use list
> comprehensions and locals()
> > Improved .gitignore
> > Fixed file permissions from 755 to 644
> > Removed license information from main file - now just in LICENSE file.
> > I'm feeling confident I left existing functionality intact, although it
> might be good if someone else tested it as well. If my commit breaks
> anything I'm willing to fix it ASAP. It feels wrong to use import *, but I
> didn't want to break anything.
> >
> > In another branch preserve-history tried to make changes to psonic.py
> file more visible, but unfortunately it wasn't any better.
> >
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/gkvoelkl/python-sonic/pull/17
> >
> > Commit Summary
> >
> > Fixed file permissions and added some structure. No implementation
> changes.
> > Aded test for original functionality
> > Separated samples and loops from psonic.py
> > Since changed original files, increased version number.
> > Bit by bit, trying to make the codebase comprehensible
> > simplified "play" function.
> > simplified "synth" function.
> > separated responsibilities, simplified a little, added tests for imports.
> > Placeholder for synth_server
> > Moved synth_server to separate file.
> > Fixed a bug that prevented playing samples
> > Increased version nr.
> > improved gitignore
> > removed reclessly added functionality, tried to clean up a little.
> > Tried to clean up a little.
> > File Changes
> >
> > A .gitignore (64)
> > M LICENSE (0)
> > M README.rst (0)
> > D psonic.py (950)
> > A psonic/__init__.py (1)
> > A psonic/api.py (84)
> > A psonic/effects.py (20)
> > A psonic/internals/__init__.py (0)
> > A psonic/internals/chords.py (74)
> > A psonic/internals/scales.py (87)
> > A psonic/notes.py (73)
> > A psonic/psonic.py (160)
> > A psonic/samples/__init__.py (6)
> > A psonic/samples/ambient.py (15)
> > A psonic/samples/bass.py (13)
> > A psonic/samples/drums.py (45)
> > A psonic/samples/electric.py (30)
> > A psonic/samples/guitars.py (9)
> > A psonic/samples/loops/__init__.py (14)
> > A psonic/samples/misc.py (8)
> > A psonic/samples/percussions.py (10)
> > A psonic/samples/tabla.py (31)
> > A psonic/samples/vinyl.py (8)
> > A psonic/scales.py (81)
> > A psonic/synth_server.py (111)
> > A psonic/synthesizers.py (51)
> > M python-sonic.ipynb (0)
> > M setup.py (46)
> > A tests/__init__.py (0)
> > A tests/conftest.py (0)
> > A tests/test_psonic.py (52)
> > Patch Links:
> >
> > https://github.com/gkvoelkl/python-sonic/pull/17.patch
> > https://github.com/gkvoelkl/python-sonic/pull/17.diff
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gkvoelkl/python-sonic/pull/17#issuecomment-370240661>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAcMtrNZ7TrbXTne1PtWQ4vcM60259MDks5tbBEfgaJpZM4STyEr>
> .
>
